### PR TITLE
release: prepare for release `v1.27.0`

### PR DESCRIPTION
**Pull request by flouthoc**
Prepare for release `v1.27.0`

[NO NEW TESTS NEEDED]
[NO TESTS NEEDED]

### CODE REVIEW

**Comment by flouthoc** [comment]
@nalind @TomSweeneyRedHat @rhatdan PTAL once approved and merged will cut a tag after this.

**Comment by rhatdan** [comment]
I am fine with that also.  Back port it.

**Comment by rhatdan** [comment]
Lets just update the prior release and leave this in main.

**Comment by nalind** [comment]
> I am fine with that also. Back port it.

To be clear, backport what?

**Comment by flouthoc** [comment]
Oh i see but which backports are we looking for in `1.26.x` for info we recently released `1.26.2` for a BZ fix and after that we only have one unreleased version where `c/storage` is bumped: https://github.com/containers/buildah/commits/release-1.26

**Comment by TomSweeneyRedHat** [comment]
In the 1.26 branch, I think it should have #4155 which is in progress on that branch.  After that merges, we should bump the Z version in the 1.26 branch.  Unless there are some commits that you think must get in.  There's just a lot of changes in there to add a week after feature freeze.

**Comment by rhatdan** [comment]
Containers/common and bug fixes back ported to current release.  I don't think there has been a lot of testing of this buildah release anyways.

**Comment by nalind** [comment]
> Containers/common and bug fixes back ported to current release.

#4155 currently pulls in containers/common v0.49.1 and cherry-picks https://github.com/containers/buildah/pull/4099, https://github.com/containers/buildah/pull/4071, and a small part of https://github.com/containers/buildah/pull/4048 to make the linters happy.  Which other bug fixes, specifically, do you have in mind?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+    build: now accepts a branch and a subdirectory when the build context is a git repository.
+    build: output now shows a progress bar while pushing and pulling images
+    build: now errors out if the path to Containerfile is a directory.
+    build: support building container images on environments that are rootless and without any valid login sessions.
+    fix: `--output` now generates artifacts even if the entire build is cached.
+    fix: `--output` generates artifacts only for the target stage in multi-stage builds.
+    fix,add: now fails on a bad HTTP response instead of writing to container
+    fix,squash: never use build cache when computing the last step of the last stage
+    fix,build,run: allow reusing secret more than once in different RUN steps
+    fix: compatibility with Docker build by making its --label and --annotate options set empty labels and annotations when given a name but no `=` or label value.
```
Comment (TomSweeneyRedHat) [inline CHANGELOG.md:30]: Just for future reference.  We have generally not updated the commit descriptions in the Changelogs in the past.  Doing so makes it harder to find the commits in the history that pertain to the change.

**Comment by TomSweeneyRedHat** [comment]
I'd vote:  
#4135 stage_executor, output: generate correct artifacts for multi-stage builds and cached builds with --output
#4013 squash, layers: never use build cache when computing last step of last stage 
#4086 add: fail on bad HTTP response instead of writing to container for URL sources
#3998 run: allow re-using secret artifact from host in new RUN steps

At least those were the ones that were listed as fixes that I'm not seeing in 1.26.

As far as the testing goes, QE has actually run several test runs already, both against Buildah and then also via Podman.

**Comment by lukasmrtvy** [comment]
Is possible to cherrypick also https://github.com/containers/buildah/pull/4064 ? Thanks

**Comment by flouthoc** [comment]
@lukasmrtvy Oh so sorry but `v1.26.3` is already out and since it was a minor it only included fixes and dependency bumps, discussion for `v1.27.0` is going here https://github.com/containers/buildah/discussions/4161

File: CHANGELOG.md (hunk 1 above)
Comment (flouthoc) [reply CHANGELOG.md:33]: Sure will take care from next time, I modified some of the messages so it becomes easier for end-user to understand the added feature but I agree it makes it difficult to go to exact commit.

**Comment by flouthoc** [comment]
CI failure is issue with upstream and is issue with all the PR's,, being discussed here: https://github.com/containers/buildah/pull/4170#issuecomment-1207905624

File: define/types.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ const (
 	Package = "buildah"
 	// Version for the Package.  Bump version in contrib/rpm/buildah.spec
 	// too.
-	Version = "1.27.0"
```
Comment (TomSweeneyRedHat) [inline define/types.go]: Don't we need a change in buildah.spec too?

**Comment by TomSweeneyRedHat** [comment]
Never mind, I missed that https://github.com/containers/buildah/commit/198b4c360b84725fd13c81fc2984fb6b73da12aa removed the need for buildah.spec So long, old friend.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@nalind PTAL

**Comment by TomSweeneyRedHat** [comment]
And I will note for the record that after we decided to use 1.26.* for Podman v4.2, we found that Podman 1.27 had been being used in Podman v4.2 already, although, at first glance, Podman looked to be using 1.26.4-dev. 

Long story short, we are preparing v1.27.0 for Podman v4.2 vendoring.

File: define/types.go (hunk 2 above)
Comment (flouthoc) [reply define/types.go]: Yes removed here: https://github.com/containers/buildah/commit/198b4c360b84725fd13c81fc2984fb6b73da12aa

**Comment by nalind** [comment]
LGTM

**Comment by flouthoc** [comment]
@TomSweeneyRedHat @nalind @rhatdan one `/lgtm` is also needed to get this in.

**Pull request merged**